### PR TITLE
Auto find websocket address and handle bytes

### PR DESCRIPTION
**Pull request by j-mendez**
* add valid bytes handling of resources ( allows storing correctly non ut8 to disk ).
* add websocket address connecting
* add unix chrome paths
* allow stealth pass in dynamic UA for non static content

### CODE REVIEW

**Comment by j-mendez** [comment]
Feel free to make adjustments to the PR to fit the needs of the crate.

File: src/browser.rs [hunk 1]
```diff
@@ -51,10 +51,51 @@ pub struct Browser {
...
+    user_agent: String,
+    #[serde(rename = "V8-Version")]
+    /// The v8 engine version
+    v8_version: String,
+    #[serde(rename = "WebKit-Version")]
+    /// Webkit version
+    webkit_version: String,
+    #[serde(rename = "webSocketDebuggerUrl")]
+    /// Remote debugging address
+    web_socket_debugger_url: String
```
Comment (mattsse) [inline src/browser.rs:74]: please make fields pub

File: src/browser.rs [hunk 2]
```diff
@@ -51,10 +51,51 @@ pub struct Browser {
...
+            .await
+            .unwrap();
+
+            let socketaddr = req.remote_addr().unwrap();
+            let connection: BrowserConnection = serde_json::from_slice(&req.bytes().await.unwrap_or_default()).unwrap_or_default();
+
+            if !connection.web_socket_debugger_url.is_empty() {
+                // prevent proxy interfaces from returning local ips to connect to the exact machine
+                debug_ws_url = connection.web_socket_debugger_url.replace("127.0.0.1", &socketaddr.ip().to_string());
+            }
```
[the remark this answers is not in the document]
Comment (j-mendez) [reply src/browser.rs:96]: This allows a url to find the websocket debugging port to connect to without knowing it beforehand.

**Review by mattsse — APPROVED** [review]
ty, sorry for the delay.

I changed my mind on this and combined the connect functions, checking for HTTP is reasonable, and less confusing than having to separate functions

**Comment by mattsse** [comment]
superseded by #188 because can't push manual changes to main

**Pull request closed without merging**